src/main/java/com/jwplayer/southpaw/util/ByteArraySet.java

```diff
@@ -219,59 +222,104 @@ public static Chunk deserializeCompact(byte[] bytes, int from, int to) {
-                                        break;
-                                    }
-                                    index += baSize;
-                                }
-                            }
-                            if(max.equals(byteArray)) {
-                                index = 0;
-                                while(index < size) {
-                                    baSize = (int) bytes[index];
-                                    index++;
-                                    if(baSize != 0) {
-                                        max = new ByteArray(Arrays.copyOfRange(bytes, index, index + baSize));
-                                    }
-                                    index += baSize;
-                                }
-                            }
+                        int comparison = valToRemove.compareTo(bytes, index + 1, entrySize);
+                        if(comparison < 0) {
+                            // valToRemove is less than the current index, so move to the next valToRemove
+                            break;
+                        } else if(comparison == 0) {
+                            Arrays.fill(bytes, index, index + entrySize + 1, (byte) 0);
+                            entries--;
+                            index += entrySize + 1;
+                            valsRemoved = true;
+                            break;
+                        } else {
+                            // valToRemove is greater than the current index, so move to the next index
+                            index += entrySize + 1;
                         }
-                        return true;
                     }
                 }
-                index += baSize;
             }
-            return false;
+            if(valsToRemove.contains(min) || valsToRemove.contains(max)) {
+                resetMinAndMax();
+            }
+
+            return valsRemoved;
+        }
+
+        /**
+         * Searches for and resets the min and max entries starting at index 0
+         */
+        public void resetMinAndMax() {
+            if(entries == 0) {
+                max = null;
+                min = null;
+                size = 0;
+            } else {
+                int entrySize = 0;
+                int index = 0;
+                while (index < size) {
+                    entrySize = bytes[index];
+                    if (entrySize != 0) {
+                        min = new ByteArray(Arrays.copyOfRange(bytes, index + 1, index + entrySize + 1));
+                        break;
+                    }
+                    index++;
+                }
+                Integer maxIndex = null;
+                while (index < size) {
+                    entrySize = bytes[index];
+                    if (entrySize != 0) {
+                        maxIndex = index;
+                        index+= entrySize;
+                    }
+                    index++;
+                }
```

Comment:
this is doing two iterations over the same byte array. Can this be combined into one loop?

Reply:
It's not actually doing two iterations. The second loop picks up where the first one left off. But it could be combined.

Reply:
Oh interesting! That wasn't clear that it was picking up where it left off. Might make more sense for future people reading the code to combine

Reply:
Yeah, that's totally understandable! I've got another commit coming with more tests. I can throw that change in there with it.